### ISSUE TITLE
Bluetooth support missing on Pi 3

### ISSUE THREAD

**Issue by snakeX79** [maxnet/berryboot#273]
Is possible to implement integrated bluetooth support for Raspberry Pi 3 ?
In  berryboot-20160313-pi2-pi3 it's missing. (Device is not available).

Thanks.

**Comment by maxnet** [comment]
No plans for it in Berryboot menu itself atm.
Should work in the Linux distributions though. (might need to do `sudo apt-get update ; sudo apt-get upgrade` in Raspbian)

**Comment by snakeX79** [comment]
Thanks. But it did not help. After update / upgrade  in Raspbian still reports "Device is not available: No such device" when I typed the command "hcitool scan".

**Comment by maxnet** [comment]
Seems one needs to install the pi-bluetooth package first.

```
pi@raspberrypi:~ $ sudo apt-get install pi-bluetooth 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following extra packages will be installed:
  bluez bluez-firmware
The following NEW packages will be installed:
  bluez bluez-firmware pi-bluetooth
0 upgraded, 3 newly installed, 0 to remove and 0 not upgraded.
Need to get 688 kB of archives.
After this operation, 2,890 kB of additional disk space will be used.
Do you want to continue? [Y/n] 
Get:1 http://archive.raspberrypi.org/debian/ jessie/main bluez armhf 5.23-2+rpi1 [595 kB]
Get:2 http://archive.raspberrypi.org/debian/ jessie/main bluez-firmware all 1.2-3+rpi1 [89.6 kB]
Get:3 http://archive.raspberrypi.org/debian/ jessie/main pi-bluetooth armhf 0.1.0 [3,182 B]
Fetched 688 kB in 0s (901 kB/s)     
Selecting previously unselected package bluez.
(Reading database ... 126324 files and directories currently installed.)
Preparing to unpack .../bluez_5.23-2+rpi1_armhf.deb ...
Unpacking bluez (5.23-2+rpi1) ...
Selecting previously unselected package bluez-firmware.
Preparing to unpack .../bluez-firmware_1.2-3+rpi1_all.deb ...
Unpacking bluez-firmware (1.2-3+rpi1) ...
Selecting previously unselected package pi-bluetooth.
Preparing to unpack .../pi-bluetooth_0.1.0_armhf.deb ...
Unpacking pi-bluetooth (0.1.0) ...
Processing triggers for systemd (215-17+deb8u3) ...
Processing triggers for dbus (1.8.20-0+deb8u1) ...
Processing triggers for man-db (2.7.0.2-5) ...
Setting up bluez (5.23-2+rpi1) ...
Setting up bluez-firmware (1.2-3+rpi1) ...
Setting up pi-bluetooth (0.1.0) ...
Processing triggers for dbus (1.8.20-0+deb8u1) ...
Processing triggers for systemd (215-17+deb8u3) ...
pi@raspberrypi:~ $ sudo hcitool scan
Scanning ...
        BC:F5:AC:9E:xx:xx       Nexus 5
```

**Comment by snakeX79** [comment]
There is no progress.

pi@raspberrypi:~ $ sudo apt-get install pi-bluetooth 
Reading package lists... Done
Building dependency tree  
Reading state information... Done
The following extra packages will be installed:
  bluez-firmware
The following NEW packages will be installed:
  bluez-firmware pi-bluetooth
0 upgraded, 2 newly installed, 0 to remove and 0 not upgraded.
Need to get 92.8 kB of archives.
After this operation, 240 kB of additional disk space will be used.
Do you want to continue? [Y/n] Y
Get:1 http://archive.raspberrypi.org/debian/ jessie/main bluez-firmware all 1.2-3+rpi1 [89.6 kB]
Get:2 http://archive.raspberrypi.org/debian/ jessie/main pi-bluetooth armhf 0.1.0 [3,182 B]
Fetched 92.8 kB in 0s (243 kB/s)  
Selecting previously unselected package bluez-firmware.
(Reading database ... 127666 files and directories currently installed.)
Preparing to unpack .../bluez-firmware_1.2-3+rpi1_all.deb ...
Unpacking bluez-firmware (1.2-3+rpi1) ...
Selecting previously unselected package pi-bluetooth.
Preparing to unpack .../pi-bluetooth_0.1.0_armhf.deb ...
Unpacking pi-bluetooth (0.1.0) ...
Setting up bluez-firmware (1.2-3+rpi1) ...
Setting up pi-bluetooth (0.1.0) ...
pi@raspberrypi:~ $ sudo hcitool scan
Device is not available: No such device

**Comment by snakeX79** [comment]
Hello,

It works ! 
After reboot.
Thanks. :-)

**Comment by snakeX79** [comment]
Hello again...

In a RetroPie it doesn't work to
I updated sudo "apt-get update" and upgraded "sudo apt-get upgrade" (I have raspbian Jessie)
And then -
pi@retropie ~ $ sudo apt-get install pi-bluetooth
Reading package lists... Done
Building dependency tree  
Reading state information... Done
E: Unable to locate package pi-bluetooth

I tried next install -
pi@retropie ~ $ sudo apt-get install bluetooth
Reading package lists... Done
Building dependency tree  
Reading state information... Done
bluetooth is already the newest version.
The following packages were automatically installed and are no longer required:
  libts-0.0-0 libts-dev tsconf
Use 'apt-get autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.

After reboot I tried -
pi@retropie ~ $ hcitool scan
Device is not available: No such device

**Comment by maxnet** [comment]
> I updated sudo "apt-get update" and upgraded "sudo apt-get upgrade" (I have raspbian Jessie)

Are you sure you have Jessie?
Think you may have an older version based on Wheezy.

Try again with Retropie 3.6

**Comment by snakeX79** [comment]
Yes Im sure.

  .~~.   .~~.    Monday, 21 March 2016, 11:43:26 am CET
  '. \ ' ' / .'   Linux 4.1.19v7-aufs armv7l GNU/Linux
   .~ .~~~..~.  
  : .~.'~'.~. :   Filesystem      Size  Used Avail Use% Mounted on
 ~ (   ) (   ) ~  none             59G   28G   29G  50% /
( : '~'.~.'~' : ) Uptime.............: 0 days, 00h06m46s
 ~ .~       ~. ~  Memory.............: 553740kB (Free) / 751776kB (Total)
  (   |   |   )   Running Processes..: 123
  '~         ~'   IP Address.........: 192.168.1.23 
    _--~-~--_     Temperature........: CPU: 46°C/114°F GPU: 46°C/114°F
                  The RetroPie Project, http://www.petrockblock.com

pi@retropie ~ $ cat /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 8 (jessie)"
NAME="Raspbian GNU/Linux"
VERSION_ID="8"
VERSION="8 (jessie)"
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"

**Comment by maxnet** [comment]
If you do `sudo apt-get update` it also prints that it is fetching the repository files for jessie?
(to make sure your sources.list isn't screwed up)

**Comment by snakeX79** [comment]
pi@retropie ~ $ sudo apt-get update
Get:1 http://mirrordirector.raspbian.org jessie InRelease [15.0 kB]  
Hit http://archive.raspberrypi.org wheezy InRelease  
Hit http://raspberrypi.collabora.com wheezy InRelease  
Hit http://archive.raspberrypi.org wheezy/main armhf Packages  
Get:2 http://mirrordirector.raspbian.org jessie/main armhf Packages [8,963 kB]
Hit http://raspberrypi.collabora.com wheezy/rpi armhf Packages
Ign http://archive.raspberrypi.org wheezy/main Translation-en_GB  
Ign http://archive.raspberrypi.org wheezy/main Translation-en  
Ign http://raspberrypi.collabora.com wheezy/rpi Translation-en_GB
Ign http://raspberrypi.collabora.com wheezy/rpi Translation-en
Get:3 http://mirrordirector.raspbian.org jessie/contrib armhf Packages [37.5 kB]
Get:4 http://mirrordirector.raspbian.org jessie/non-free armhf Packages [70.3 kB]
Get:5 http://mirrordirector.raspbian.org jessie/rpi armhf Packages [1,356 B]  
Ign http://mirrordirector.raspbian.org jessie/contrib Translation-en_GB  
Ign http://mirrordirector.raspbian.org jessie/contrib Translation-en  
Ign http://mirrordirector.raspbian.org jessie/main Translation-en_GB  
Ign http://mirrordirector.raspbian.org jessie/main Translation-en
Ign http://mirrordirector.raspbian.org jessie/non-free Translation-en_GB
Ign http://mirrordirector.raspbian.org jessie/non-free Translation-en
Ign http://mirrordirector.raspbian.org jessie/rpi Translation-en_GB
Ign http://mirrordirector.raspbian.org jessie/rpi Translation-en
Fetched 9,087 kB in 55s (165 kB/s)
Reading package lists... Done

**Comment by maxnet** [comment]
Strange, because it does work fine here...

```
Hit http://archive.raspberrypi.org wheezy InRelease
```

You are combining wheezy with jessie packages?

**Comment by snakeX79** [comment]
I don't understand it so much. But I tried New instalation of Raspbian from berryboot and installed Retropie via script and bluetooth. It works fine now. So I solved it.
Thank you very much for your help :-)

**Comment by notoriousdds** [comment]
SOLVED:

I realize this thread is over a year old, but...

I had this same problem.  I had a highly tweaked raspbian 'wheezy' image that I upgraded to 'jessie' so that I could use in on my new Pi 3.  I didn't want to start over with a fresh jessie install.  However, I couldn't get the bluetooth to work for anything.  It was this post that led me to the answer...

It appears that the upgrade process only changes some of the package sources to 'jessie'.  The tell tale sign is if you're running 'jessie' and when attempting to install the pi-bluetooth package, you get the following error:

`Unable to locate package pi-bluetooth`

The solution was to make sure that **BOTH** of the following lists were calling for 'jessie' packages:

> /etc/apt/sources.list
 /etc/apt/sources.list.d/raspi.list

In my case, only the sources.list had been updated but my raspi.list was still using 'wheezy'.

I simply changed the word 'wheezy' to 'jessie' and then:

`sudo apt update && sudo apt upgrade`

After about 20 minutes, the upgrade will finish.  

Reboot and Voilá
